File: app/src/main/java/com/eslimaf/mobilechallenge/infrastructure/waveapi/WaveApi.java
package com.eslimaf.mobilechallenge.infrastructure.waveapi;

import com.eslimaf.mobilechallenge.model.Product;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface WaveApi {
    String WAVEAPI_BASE_URL = "https://api.waveapps.com";

    @GET("businesses/{businessId}/products")
    Call<List<Product>> getProductsByBusinessId(@Path("businessId") String businessId);
}

File: WaveExample/app/src/main/kotlin/com/derekcsm/wave_example/_api/WaveApi.kt
package com.derekcsm.wave_example._api

import com.derekcsm.wave_example._model.Product
import retrofit2.http.GET
import retrofit2.http.Path
import rx.Observable

interface WaveApi {

  @GET("businesses/{business_id}/products")
  fun getProducts(@Path("business_id") businessId: String): Observable<List<Product>>
}